### PR TITLE
Add pybind11 to thirdparty

### PR DESCRIPTION
**Pull request by chuckatkins**

### CODE REVIEW

**Comment by chuckatkins** [comment]
Hold off a minute on this, I need to rebase back to the right point.

**Comment by williamfgc** [comment]
@chuckatkins let me know when the PR is ready

**Comment by chuckatkins** [comment]
It's good to go.  I'm fixing the CMake build not too so that will be there shortly.

File: source/adios2/ADIOSConfig.h.in [hunk 1]
```diff
@@ -42,4 +42,7 @@
 /* CMake Option: ADIOS_USE_DataMan=TRUE */
 #cmakedefine ADIOS2_HAVE_DATAMAN
 
+/* CMake Option: ADIOS_USE_Python=TRUE */
+#cmakedefine ADIOS2_HAVE_Python
```
Comment (williamfgc) [inline source/adios2/ADIOSConfig.h.in:46]: Should we distinguish between Python 2.7.x and 3?

**Review by williamfgc — CHANGES_REQUESTED** [review]
Should we distinguish between Python 2.7.x and 3?

File: source/adios2/ADIOSConfig.h.in (hunk 1 above)
Comment (chuckatkins) [reply source/adios2/ADIOSConfig.h.in:46]: Does it matter?

Comment (williamfgc) [reply source/adios2/ADIOSConfig.h.in:46]: I was thinking if we have two python versions present. That's how I tested Python 2.7.x and 3.5 in the same system. I am assuming most users will have Python 2.7.x as default in /usr/bin/python

**Pull request merged**